tests/unit/sagemaker/workflow/test_pipeline_graph.py

```diff
@@ -0,0 +1,92 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pytest
+
+from mock import Mock
+
+from sagemaker.workflow.pipeline import Pipeline, PipelineGraph
+from tests.unit.sagemaker.workflow.helpers import ordered, CustomStep
+
+
+@pytest.fixture
+def sagemaker_session_mock():
+    session_mock = Mock()
+    session_mock.default_bucket = Mock(name="default_bucket", return_value="s3_bucket")
+    return session_mock
+
+
+@pytest.fixture
+def role_arn():
+    return "arn:role"
+
+
+def test_pipeline_duplicate_step_name(sagemaker_session_mock):
```

Comment:
A couple of more scenarios:
* duplicate_step_name_in_condition_step
* duplicate_step_name_in_step_collection